### PR TITLE
Update BASH_ENV code examples

### PR DESCRIPTION
**Pull request by kbatuigas**
# Description
Place double quotes around env var names in BASH_ENV examples
Also add comment explaining why `source`  might included in code example

# Reasons
If any of the variables contain spaces or special characters in them, adding double-quotes in the right places will ensure that any spaces or special characters are properly escaped.
[Jira ticket](https://circleci.atlassian.net/browse/DOCTEAM-559)

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by devinmarieb** [comment]
@kbatuigas have these been checked by engineering? I'm under the impression you would need to escape one set of double quotes in an example like this where there are double quotes wrapped in double quotes:
`echo "export __BUILD_VERSION=\"$(cat version.txt)\"" >> "$BASH_ENV"`

**Comment by devinmarieb** [comment]
@kbatuigas Have these been checked by engineering? It looks good, but I don't know that I have enough knowledge to know if quotes wrapped in quotes are being escaped in the right places, etc.

File: jekyll/_cci2/sample-config.md [hunk 1]
```diff
@@ -736,7 +736,7 @@ jobs:
             - run:
                   name: Setup __BUILD_VERSION envvar
                   command: |
-                      echo "export __BUILD_VERSION=\"$(cat version.txt)\"" >> $BASH_ENV
+                      echo 'export __BUILD_VERSION=\"$(cat version.txt)\"' >> "$BASH_ENV"
```
[the remark this answers is not in the document]
Comment (EricRibeiro) [reply jekyll/_cci2/sample-config.md:739]: ^ same goes for all other instances in this file.

**Review by EricRibeiro — APPROVED** [review]
Good job!

**Review by devinmarieb — APPROVED** [review]

**Pull request merged**